forage-android/src/main/java/com/joinforage/forage/android/ForageSDK.kt

```diff
@@ -12,36 +11,41 @@ import com.joinforage.forage.android.network.TokenizeCardService
 import com.joinforage.forage.android.network.data.CapturePaymentRepository
 import com.joinforage.forage.android.network.data.CheckBalanceRepository
 import com.joinforage.forage.android.network.model.ForageApiResponse
-import com.joinforage.forage.android.ui.ForagePANEditText
-import com.joinforage.forage.android.ui.ForagePINEditText
+import com.joinforage.forage.android.ui.ForageContext
+import com.joinforage.forage.android.ui.InternalForageElement
 import java.util.UUID
 
 /**
  * Singleton responsible for implementing the SDK API
  */
-object ForageSDK : ForageSDKApi {
-    private val logger = Log.getInstance()
+class ForageSDK : ForageSDKInterface {
 
-    // TODO: this should be a Config argument that uses the builder pattern
-    override suspend fun tokenizeEBTCard(
-        merchantAccount: String,
-        foragePanEditText: ForagePANEditText,
-        bearerToken: String,
-        customerId: String,
-        reusable: Boolean
-    ): ForageApiResponse<String> {
+    private fun _getForageContextOrThrow(element: InternalForageElement): ForageContext {
+        val context = element.getForageContext()
+        // TODO: create a custom Exception instead of using IllegalArgumentException
+        return context ?: throw IllegalArgumentException(
+            "You need to call element.setForageContext(forageContext: ForageContext) on a ForageElement before you can call submit."
+        )
+    }
```

Comment:
this would crash the client's app if they don't wrap the calls to `tokenizeEbtCard` in a try/cratch.

What are your thoughts on documenting/annotating that that `tokenizeEbtCard`, `checkBalance` method throws `IllegalArgumentException` (or whatever it ends up being named using Kotlin / Android conventions.

Reply:
100% that needs to happen! Completely agree!

Reply:
prosing `MissingForageConfigException` as the name for the exception and added docstrings to all public surface in [5dcb8b3](https://github.com/teamforage/forage-android-sdk/pull/97/commits/5dcb8b36a2906318b3650c39eb8fc37a85139a3b). Will mark as resolved but lmk what you think of the docstrings!